### PR TITLE
[v4r0] Pylint test: fix finding of webapp python files

### PR DESCRIPTION
**Pull request by andresailer**
This should fix the finding of the python files for the pylint test.
And if correct there are a number of errors

### CODE REVIEW

**Review by fstagni — APPROVED** [review]

**Comment by andresailer** [comment]
Errors found:
```
************* Module WebAppDIRAC.WebApp.handler.ConfigurationManagerHandler
WebApp/handler/ConfigurationManagerHandler.py:86: [E1136(unsubscriptable-object), ConfigurationManagerHandler.__getRemoteConfiguration] Value 'modCfg.getCFG()['DIRAC']' is unsubscriptable
WebApp/handler/ConfigurationManagerHandler.py:87: [E1136(unsubscriptable-object), ConfigurationManagerHandler.__getRemoteConfiguration] Value 'modCfg.getCFG()['DIRAC']' is unsubscriptable
************* Module WebAppDIRAC.WebApp.handler.SiteSummaryHandler
WebApp/handler/SiteSummaryHandler.py:47: [E1101(no-member), SiteSummaryHandler.web_getSelectionData] Instance of 'set' has no 'sort' member
WebApp/handler/SiteSummaryHandler.py:87: [E1136(unsubscriptable-object), SiteSummaryHandler.web_action] Value 'requestParams['action']' is unsubscriptable
************* Module WebAppDIRAC.WebApp.handler.ProxyUploadHandler
WebApp/handler/ProxyUploadHandler.py:4: [E0401(import-error), ] Unable to import 'DIRAC.Core.Security.X509Chain'
************* Module WebAppDIRAC.WebApp.handler.RegistryManagerHandler
WebApp/handler/RegistryManagerHandler.py:66: [E1136(unsubscriptable-object), RegistryManagerHandler.__getRemoteConfiguration] Value 'modCfg.getCFG()['DIRAC']' is unsubscriptable
WebApp/handler/RegistryManagerHandler.py:67: [E1136(unsubscriptable-object), RegistryManagerHandler.__getRemoteConfiguration] Value 'modCfg.getCFG()['DIRAC']' is unsubscriptable
************* Module WebAppDIRAC.WebApp.handler.ResourceSummaryHandler
WebApp/handler/ResourceSummaryHandler.py:46: [E1101(no-member), ResourceSummaryHandler.web_getSelectionData] Instance of 'set' has no 'sort' member
WebApp/handler/ResourceSummaryHandler.py:177: [E1136(unsubscriptable-object), ResourceSummaryHandler.web_action] Value 'requestParams['action']' is unsubscriptable
************* Module WebAppDIRAC.WebApp.handler.SystemAdministrationHandler
WebApp/handler/SystemAdministrationHandler.py:350: [E1126(invalid-sequence-index), SystemAdministrationHandler.web_componentAction] Sequence index is not an int, slice, or instance with __index__
WebApp/handler/SystemAdministrationHandler.py:353: [E1126(invalid-sequence-index), SystemAdministrationHandler.web_componentAction] Sequence index is not an int, slice, or instance with __index__
WebApp/handler/SystemAdministrationHandler.py:584: [E1121(too-many-function-args), SystemAdministrationHandler.sendMail] Too many positional arguments for constructor call
************* Module WebAppDIRAC.Lib.WebHandler
Lib/WebHandler.py:17: [E0401(import-error), ] Unable to import 'DIRAC.Core.Security.X509Chain'
Lib/WebHandler.py:42: [E1003(bad-super-call), WErr.__str__] Bad first argument 'HTTPError' given to super()
Lib/WebHandler.py:88: [E0602(undefined-variable), WebHandler.threadTaskOld] Undefined variable 'deprecated'
Lib/WebHandler.py:153: [E1101(no-member), WebHandler.__init__] Instance of 'str' has no 'match' member
Lib/WebHandler.py:273: [E1101(no-member), WebHandler.getUserProperties] Instance of 'dict' has no 'properties' member
```

**Comment by fstagni** [comment]
Heh... can you go ahead and correct them?

**Comment by andresailer** [comment]
I will try a fix, and then we can see which fixes are correct

File: WebApp/handler/SystemAdministrationHandler.py [hunk 1]
```diff
@@ -581,7 +581,7 @@ def sendMail(self, sendDict=None, title=None, body=None, fromAddress=None):
     sentSuccess = list()
     sentFailed = list()
     gLogger.debug("Initializing Notification client")
-    ntc = NotificationClient(lambda x, timeout: RPCClient(x, timeout=timeout, static=True))
```
Comment (andresailer) [inline WebApp/handler/SystemAdministrationHandler.py:584]: The NotificationClient no longer takes a functor
https://github.com/DIRACGrid/DIRAC/commit/f719136fd27e6fb7063e8aa1a44ba500e6175207

**Comment by andresailer** [comment]
I don't understand if this is a bug or something pylint cannot resolve:
```
Lib/WebHandler.py:153: [E1101(no-member), WebHandler.__init__] Instance of 'str' has no 'match' member
```
https://github.com/DIRACGrid/WebAppDIRAC/blob/60254d791bb31b529b4f37c457c87e0879a5e833/Lib/WebHandler.py#L153

----

This function is not used ? (getData returns a dictionary, so this should fail...)
https://github.com/DIRACGrid/WebAppDIRAC/blob/60254d791bb31b529b4f37c457c87e0879a5e833/Lib/WebHandler.py#L273

**Comment by andresailer** [comment]
Fixed all the errors except the two mentioned above

**Review by fstagni — APPROVED** [review]

**Comment by andresailer** [comment]
What about those errors https://github.com/DIRACGrid/WebAppDIRAC/pull/334#issuecomment-510569423 ?

**Comment by zmathe** [comment]
Please change to PATH_RE = None https://github.com/DIRACGrid/WebAppDIRAC/blob/integration/Lib/WebHandler.py#L79

**Comment by fstagni** [comment]
It still fails.

**Comment by andresailer** [comment]
The only thing that still fails is this:
`Lib/WebHandler.py:274: [E1101(no-member), WebHandler.getUserProperties] Instance of 'dict' has no 'properties' member`

**Comment by zmathe** [comment]
it seams the method is not used. In addition the return value is a dict.
Could you please change to:
```python
return self.__sessionData.getData()['properties']
```

**Comment by andresailer** [comment]
I think then we should just remove the function, the getData() dictionary does not have a `['properties']` entry
https://github.com/DIRACGrid/WebAppDIRAC/blob/60254d791bb31b529b4f37c457c87e0879a5e833/Lib/SessionData.py#L115-L134

**Comment by zmathe** [comment]
yes, you can remove it. The method is not used.

**Comment by andresailer** [comment]
All green now

**Comment by zmathe** [comment]
@atsareg Please include also WebApp PRs to the next release

**Pull request merged**